### ISSUE TITLE
[RTK v2.0] createApi (React) exceeds the maximum length the compiler will serialize

### ISSUE THREAD

**Issue by eric-crowell** [reduxjs/redux-toolkit#3591]
Version: `2.0.0-beta.0`
Hello, I'm having an issue inferring types when using `createApi` from `@reduxjs/toolkit/query/react`.

I get this issue with a very minimal setup.

[Code Sandbox link](https://codesandbox.io/p/sandbox/rtk-type-portability-j99lyh?file=%2Fsrc%2Frtk%2FmyApiReact.ts%3A6%2C18)
```typescript
import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";

/**
 * `myApiReact` defaults to the `any` type producing the TypeScript error....
 * The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed.ts(7056)
 */ 
export const myApiReact = createApi({
  reducerPath: "myApiReact",
  baseQuery: fetchBaseQuery({
    baseUrl: "/",
  }),
  endpoints: (builder) => ({
    getData: builder.query<{}, string>({
      query: (id) => `data/${id}`,
    }),
  }),
});
```

I can't find the cause of it yet. Seems like there's some type recursion happening somewhere in `enhanceEndpoints` when the react module is used? Maybe someone has a better clue.

**Comment by markerikson** [comment]
Two questions:

- Does that happen with 1.9?  Don't think we changed any RTKQ types for 2.0
- Does that happen if you change the `{}` return type to any other type?

**Comment by eric-crowell** [comment]
Oops! I apologize. I should have added the necessary TypeScript configurations needed to get this issue.

```typescript
// tsconfig.json
{
  // ...
  "moduleResolution": "Node16", // Or "NodeNext",
  "declaration": true,
  // ...
}

It only occurs with `Node16` as the `moduleResolution` AND if one is exporting declarations; otherwise, there's no issue.

Now that I'm writing it out, it might be similar to #3568 and there's some type that needs to be exported now for the new ES6 module stuff.

No changes on using a different type other than `{}`.

**Comment by eric-crowell** [comment]
I think this pull request resolves the issue: #3592

It did fix it for me.
